### PR TITLE
[Wizard] Allow opening help messages from the app

### PR DESCRIPTION
**Pull request by tokland**
Required by https://github.com/EyeSeeTea/vaccination-app/pull/238

Implementation:

- Make `DialogButton` opening available as a prop (`initialIsOpen`).
- Update `Wizard` to pass `step.helpDialogIsInitialOpen` to `DialogButton`.

### CODE REVIEW

**Review by SferaDev — APPROVED** [review]
Code looks good to me, tested with vaccination-app and the feature works fine. Also tested the DialogButton in metadata-sync and doesn't seem we have any side effect there.

File: src/wizard/Wizard.jsx [hunk 1]
```diff
@@ -213,7 +222,7 @@ class Wizard extends React.Component {
                                 {step.label}
                             </StepButton>
 
-                            {step.help && step === currentStep ? <Help step={step} /> : null}
```
Comment (SferaDev) [inline src/wizard/Wizard.jsx:216]: Help button is always rendered even when ```step.help``` is undefined.

Proposal: ``` {step.help && <Help step={step} currentStepKey={currentStepKey} /> }```

image

image


```
Warning: Failed prop type: The prop `contents` is marked as required in `DialogButton`, but its value is `undefined`.
    in DialogButton
    in Unknown (created by Wizard)
    in div (created by Step)
    in Step (created by WithStyles(Step))
    in WithStyles(Step) (created by Wizard)
    in div (created by Paper)
    in Paper (created by WithStyles(Paper))
    in WithStyles(Paper) (created by Stepper)
    in Stepper (created by WithStyles(Stepper))
    in WithStyles(Stepper) (created by Wizard)
    in div (created by Wizard)
    in Wizard (created by WithStyles(Wizard))
    in WithStyles(Wizard) (created by withSnackbarWithStyles(Wizard))
    in withSnackbarWithStyles(Wizard) (at SyncRulesWizard.jsx:146)
    in SyncRulesWizard (created by Context.Consumer)
    in withRouter(SyncRulesWizard) (at Root.jsx:60)
    in Route (at Root.jsx:58)
    in Switch (at Root.jsx:25)
    in Root (at App.jsx:61)
    in SnackbarProvider (at App.jsx:60)
    in div (at App.jsx:59)
    in SnackbarProvider (at App.jsx:55)
    in MuiThemeProvider (at App.jsx:54)
    in MuiThemeProviderOld (at App.jsx:53)
    in JssProvider (at App.jsx:52)
    in App (at src/index.js:77)
    in Router (created by HashRouter)
    in HashRouter (at src/index.js:76
```

**Review by SferaDev — CHANGES_REQUESTED** [review]
Found a side-effect

File: src/wizard/Wizard.jsx (hunk 1 above)
Comment (tokland) [reply src/wizard/Wizard.jsx:216]: > Help button is always rendered even when step.help is undefined

Indeed! Your proposal looks fine to me.

**Review by SferaDev — APPROVED** [review]

**Pull request merged**